lib/promoted/ruby/client/options.rb

```diff
@@ -0,0 +1,215 @@
+          @client_log_timestamp    = args[:client_log_timestamp] || Time.now.to_i
+          @event_api_timestamp     = args[:event_api_timestamp]
+          @request_id              = args[:request_id] || SecureRandom.uuid
+        end
+
+        def tranlate_args(args)
+          args.transform_keys(&:to_underscore).transform_keys(&:to_sym)
+        rescue => e
+          raise 'Unable to parse args. Please pass correct arguments. Must be JSON'
+        end
+
+        def validate_request_paramas
+
+        end
+
+        def request
+          @request
+        end
+
+        def event_api_timestamp
+          @event_api_timestamp
+        end
+
+        def client_log_timestamp
+          @client_log_timestamp
+        end
+
+        def platform_id
+          @platform_id
+        end
+
+        def view_id
+          @view_id
+        end
+
+        def user_id
+          return @user_id if @user_id
+          @user_id = request.dig(:user_info, :user_id)
+          @user_id ||= request.dig('user_info', 'user_id')
+          @user_id
+        end
+
+        def session_id
+          @session_id
+        end
+
+        #A list of the response Insertions.  This list should be truncated
+        #based on limit.
+        def insertion
+          @insertion
+        end
+
+        def log_user_id
+          return @log_user_id if @log_user_id
+          @log_user_id   = request.dig(:user_info, :log_user_id)
+          @log_user_id ||= request.dig('user_info', 'log_user_id')
+          @log_user_id
+        end
+
+        #A way to turn off logging.  Defaults to true.
+        def enabled?
+          @enabled
+        end
+
+         #Performs extra dev checks.  Safer but slower.  Defaults to true.
+        def perform_checks?
+          @perform_checks
+        end
+
+
+        # Default values to use on DeliveryRequests.
+        def default_request_values
+          @default_request_values
+        end
+
+        # Required as a dependency so clients can load reduce dependency on multiple
+        # uuid libraries.
+        def uuid
+          @uuid
+        end
+
+        # Defaults to 250ms
+        def delivery_timeout_millis
+          @delivery_timeout_millis
+        end
+
+        # Defaults to 3000ms
+        def metrics_timeout_millis
+          @metrics_timeout_millis
+        end
+
+        # For testing.  Allows for easy mocking of the clock.
+        def now_millis
+          @now_millis
+        end
+
+        def only_Log
+          @only_Log
+        end
+
+        def full_insertion
+          @full_insertion
+        end
+
+        def user_info
+          {
+            user_id: user_id,
+            log_user_id: log_user_id
+          }
+        end
+
+        def timing
+          @timing = {
+            client_log_timestamp: client_log_timestamp,
+            event_api_timestamp: event_api_timestamp
+          }
+        end
+
+        def request_id
+          @request_id
+        end
+
+        def log_request_params
```

Comment:
When do you use "_params"?

Reply:
Following the convention and keeping it more readable.

Reply:
When is "_params" used though?  It's not obvious to me.